src/main/java/org/broadinstitute/hellbender/utils/read/ReadUtils.java

```diff
@@ -1120,4 +1120,57 @@ public static boolean validateExpectedSortOrder(
         return isValid;
     }
 
+
+    /**
+     * Sets a collection of GATKReads as supplemental reads of the primary read,
+     * This tool will set the isSupplemental attribute as well as the 'SA:' tag
+     * properly according to the samspec (example: SA:primaryreadname,position,strand,Cigar,mapQ,NM;...)
+     *
+     * @param primaryRead the primary read in the set of supplemental reads
+     * @param supplementalReads an arbitrarily sorted list of reads to be set as
+     *                          supplemental to eachother and the primary read
+     *
+     * @throws UserException 
+     */
+    public static void setReadsAsSupplemental(GATKRead primaryRead, List<GATKRead> supplementalReads) {
+        List<String> orderedTags = new LinkedList<>();
```

Comment:
Don't use a linked list here ( or really anywhere...)  They're so much slower for everything but adding or removing nodes in the middle.  Here you use it for random access.  If you're worried about the resizing cost set the `ArrayList(size)` when you create it since you know that based on the number of reads that were provided.